lib/list.c

```diff
@@ -48,3 +49,33 @@ void add_node_end(struct slist *head, struct slist *node)
 	node->next = NULL;
     }
 }
+
+/* if pos > length of linked list, insert at the end */ 
+void add_node_pos(struct slist **head, struct slist *node, int insert_at_pos)
+{
+    int count;
+    struct slist *pos, *prev_pos;
+    struct slist *headp = *head;
+
+    pos = prev_pos = NULL;
+
+    // fix invalid user input
+    if (insert_at_pos <= 0)
```

Comment:
Branching main again and fixing it (I merged upstream and delete the branch in the absent of review comments).